app/controllers/concerns/exception_handler.rb

```diff
@@ -0,0 +1,15 @@
+# app/controllers/concerns/exception_handler.rb
+module ExceptionHandler
+  # provides the more graceful `included` method
+  extend ActiveSupport::Concern
+
+  included do
+    rescue_from ActiveRecord::RecordNotFound do |e|
+      json_response({ message: e.message }, status: :not_found)
+    end
+
+    rescue_from ActiveRecord::RecordInvalid do |e|
+      json_response({ message: e.message }, status: :unprocessable_entity)
+    end
```

Comment:
This design pattern is also for a long-term perspective. So all Active records can reuse this helper when it comes to handling expectations. It can also be updated when the need comes.